proxy/pathpatch_test.go

```diff
@@ -0,0 +1,51 @@
+package proxy
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/zalando/skipper/dataclients/routestring"
+	"github.com/zalando/skipper/filters/builtin"
+	"github.com/zalando/skipper/routing"
+)
+
+func testPatch(t *testing.T, title string, f Flags, expectedStatus int) {
+	t.Run(title, func(t *testing.T) {
+		dc, err := routestring.New(`Path("/foo%2Fbar") -> status(200) -> <shunt>`)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		rt := routing.New(routing.Options{
+			SignalFirstLoad: true,
+			FilterRegistry:  builtin.MakeRegistry(),
+			DataClients:     []routing.DataClient{dc},
+		})
+		defer rt.Close()
+
+		p := WithParams(Params{Routing: rt, Flags: f})
+		defer p.Close()
+
+		s := httptest.NewServer(p)
+		defer s.Close()
+
+		<-rt.FirstLoad()
+
+		rsp, err := http.Get(s.URL + "/foo%2Fbar")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer rsp.Body.Close()
+
+		if rsp.StatusCode != expectedStatus {
+			t.Error()
+		}
+	})
+}
+
+func TestPathPatch(t *testing.T) {
+	testPatch(t, "not patched", FlagsNone, http.StatusNotFound)
+	testPatch(t, "patched", PatchPath, http.StatusOK)
```

Comment:
Can you also test a utf-8 character, which should be allowed in the path?
Because you work on []byte slices later and I want to make sure that this will work.

Reply:
Sure, I can. Why it will work, though, is that we only compare bytes in the ASCII range ('%', and '%xx'), and we exit on the first difference.

Reply:
just tried with /世%2F界. The rfc.PatchPath() function works as expected, but the test in the proxy doesn't. The reason is that in this case the http.Request.URL.RawPath remains empty. I'll investigate why.

Reply:
We agreed on ignoring it on the proxy and drop the suggested test, but have it tested in rfc filter package.